eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -698,6 +716,155 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
           goto(NORMAL) using d.copy(channelUpdate = channelUpdate1) storing()
       }
 
+    case Event(cmd: CMD_SPLICE, d: DATA_NORMAL) =>
+      val replyTo = if (cmd.replyTo == ActorRef.noSender) sender() else cmd.replyTo
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          val parentCommitments = d.metaCommitments.main
+          if (parentCommitments.hasNoPendingHtlcsOrFeeUpdate && parentCommitments.channelFeatures.hasFeature(DualFunding)) {
```

Comment:
I think we have an issue if we have proposed htlcs:

- we sent `update_add_htlc` and thus have a `CMD_SIGN` queued
- this condition will evaluate to `true`, so we move to `SpliceStatus.SpliceRequested`
- we will reject our own `CMD_SIGN` so we have a weird state with that dangling htlc?

Note that we'll have the same issue if our peer sends us `update_add_htlc` and their `commit_sig` in parallel of our `tx_init_rbf`, we will send our `revoke_and_ack` but will never send our `commit_sig`.

I'm not sure what exactly this will lead to (force-close when the splice completes because of a sig mismatch?). Maybe this is ok for a first prototype, that we will fix when adding quiescence, but we need to make sure this cannot lead to an `ERR_INFORMATION_LEAK` where we lose funds.

Reply:
I think the core issue is that `hasNoPendingHtlcsOrFeeUpdate` is far too lax. I introduced a `isIdle` in 42673dff0, which checks that we have absolutely no pending changes.